include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h

```diff
@@ -98,7 +98,7 @@ class all_view
     }
 
     // Unified function to get pointer or accessor to use inside ESIMD kernels
-    __accessor_t data()
+    __accessor_t data() const
     {
         return __m_acc;
```

Comment:
What about the case when `_AccMode` is `read` in 
`using __accessor_t = sycl::accessor<_T, 1, _AccMode, _Target, _Placeholder>;`

Reply:
Could you describe what is wrong here?

Reply:
We have discussed offline. 
The problem is related to the diagnostics of the wrong input. Namely, the necessity to notify a user if one provides sequences which do not allow both reading and writing.